distribution/test/itests/test-itests-ddf/src/test/java/ddf/test/itests/catalog/TestFederation.java

```diff
@@ -2112,6 +2144,182 @@ public void testProductDownloadWithTwoUsers() throws Exception {
                 "COMPLETE");
     }
 
+    @Test
+    public void testTwoUsersSameProductRetrySuccess() throws Exception {
+
+        String filename = "product2.txt";
+
+        CometDClient cometDClient1 = setupCometDClientWithUser(Arrays.asList(NOTIFICATIONS_CHANNEL,
+                ACTIVITIES_CHANNEL), "localhost", "localhost");
+        CometDClient cometDClient2 = setupCometDClientWithUser(Arrays.asList(NOTIFICATIONS_CHANNEL,
+                ACTIVITIES_CHANNEL), "admin", "admin");
+
+        setupCswServerForSuccess(cometDClient1, filename);
+        setupCswServerForSuccess(cometDClient2, filename);
+
+        // Verify that we get 3 notifications: 2 retrying and 1 complete.
+        // Verify that we get 9 activity messages: started, downloading, retrying, and complete
+        checkExpectations(cometDClient1, 3, 9);
+        checkExpectations(cometDClient2, 3, 9);
+
+        List<String> activities = cometDClient1.getMessagesInAscOrder(ACTIVITIES_CHANNEL);
+        boolean activityFound = foundExpectedActivity(activities, filename, "COMPLETE");
+        assertThat(activityFound, is(equalTo(true)));
+
+        activities = cometDClient2.getMessagesInAscOrder(ACTIVITIES_CHANNEL);
+        activityFound = foundExpectedActivity(activities, filename, "COMPLETE");
+        assertThat(activityFound, is(equalTo(true)));
+
+        List<String> notifications = cometDClient1.getMessagesInAscOrder(NOTIFICATIONS_CHANNEL);
+        boolean notificationFound = foundExpectedActivity(notifications, filename, "completed");
+        assertThat(notificationFound, is(equalTo(true)));
+
+        notifications = cometDClient2.getMessagesInAscOrder(NOTIFICATIONS_CHANNEL);
+        notificationFound = foundExpectedActivity(notifications, filename, "completed");
+        assertThat(notificationFound, is(equalTo(true)));
+
+    }
+
+    @Test
+    public void testTwoUsersSameProductRetryFailure() throws Exception {
+
+        String filename = "product2.txt";
+
+        CometDClient cometDClient1 = setupCometDClientWithUser(Arrays.asList(NOTIFICATIONS_CHANNEL,
+                ACTIVITIES_CHANNEL), "localhost", "localhost");
+        CometDClient cometDClient2 = setupCometDClientWithUser(Arrays.asList(NOTIFICATIONS_CHANNEL,
+                ACTIVITIES_CHANNEL), "admin", "admin");
+
+        setupCswServerForFailure(cometDClient1, filename);
+        setupCswServerForFailure(cometDClient2, filename);
+
+        // Verify that we get 4 notifications: 3 retrying and 1 complete.
+        // Verify that we get 8 activity messages: started, downloading, retrying, and complete
```

Comment:
Same reasoning as my comment above, your `setupCswServerForFailure` makes the csw stub server fail 3 times, meaning these activities probably refer to:
`started, downloading, retrying, downloading, retrying, downloading, retrying, failed`

The last retry message looks like some confusion in our cometD implementation of retries vs attempts. The first retry message says "Resource retrieval retrying after 1 bytes. Attempt 1 of 3." meaning the message that says "Attempt 3 of 3" is posted at the very end immediately before it fails. Technically the first message should be "Attempt 2 of 3" and the last one shouldn't be posted. That's a fix for an entirely different ticket, though.